src/ParameterParser.h

```diff
@@ -147,6 +148,27 @@ namespace snowcrash {
                 subject = GetFirstLine(node->children().front().text, remainingContent);
                 TrimString(subject);
 
+                // Look ahead into nested list items
+                for (MarkdownNodeIterator it = node->children().begin();
+                     it != node->children().end();
+                     ++it) {
+
+                    if (it->type == mdp::ListItemMarkdownNodeType
+                        && !it->children().empty()) {
+
+                        mdp::ByteBuffer itSubject, itRemainingContent;
+                        itSubject = GetFirstLine(it->children().front().text, itRemainingContent);
+                        TrimString(itSubject);
+
+                        if (RegexMatch(itSubject, MSONDefaultTypeSectionRegex) ||
```

Comment:
This feels brittle from the future's perspective. Adding let's say validation section in MSON parser means somebody will have to remember to add it here. It would be better if this would be reusing some part of MSON parser to identify whether there is a nested MSON section.